build_tools/scripts/update_submodules.sh

```diff
@@ -37,8 +37,29 @@ if [ -n "$git_status" ]; then
   exit 1
 fi
 
-echo "Pulling latest from remote"
-git pull
+master_url="$(git remote get-url master)"
+upstream_url="https://github.com/google/iree.git"
+if [ ${master_url} == ${upstream_url} ]
+then
+  echo "Pulling latest from remote"
+  git pull
+else
+  for remote in `git remote -v | awk -F\  '{print $1}' | sort | uniq`
+  do
+    remote_url="$(git remote get-url ${remote})"
+    if [ ${remote_url} == ${upstream_url} ]
+    then
+      upstream=${remote}
+      break
+    fi
+  done
+
+  echo "Fetching" ${upstream}
```

Comment:
Per style guide comment, put the variable expansion within the quotes. Same below.

Also, I think it's probably a good idea to abort if this isn't set, so how about adding a `?`. The style guide doesn't comment, but I think it's probably better to just default to doing that for variable expansions. Usually if you have an unset variable, you screwed up...

```suggestion
  echo "Fetching ${upstream?}"
```

Reply:
Forgot about abort if it isnt set. Ill add that.

Reply:
I dont think this is necessary any more